### DISCUSSION TITLE
Flashing does work

### DISCUSSION THREAD

**Discussion by HollaDieWaldfee2022** [jomjol/AI-on-the-edge-device#1369] [Q&A]
Hi,

I hope some can help me. Im stack at the start because the flashing does work for me.
1. I tried flashing the ESP32 with the Flashing Tool:
![grafik](https://user-images.githubusercontent.com/109627421/202549214-16b50888-7205-4725-9cfc-7c4c2a01c13d.png)

After disconnection the ESP from the PC an reconnection it (I copied all stuff to the SD-Card and edited the wifi.ini) the led unter the Camera doesnt blink an the ESP is not connection to mi wifi!

2. I tried flashing via https://jomjol.github.io/AI-on-the-edge-device

![grafik](https://user-images.githubusercontent.com/109627421/202550891-242551cb-7b05-4a86-aede-da88dc80c636.png)

![grafik](https://user-images.githubusercontent.com/109627421/202551177-0c7c38dd-481e-40fd-949f-3be530fd88a7.png)

![grafik](https://user-images.githubusercontent.com/109627421/202551600-0b9ebabd-0378-4f0c-a893-d50c5647be3c.png)

My SD-Card (32GB, FAT32, might thta be the problem?) looks like:
![grafik](https://user-images.githubusercontent.com/109627421/202551471-f85d671d-a411-4d4a-95e3-ecd66d928397.png)

But should the LED blink after flashing the device withou any SD-Card?

Best regard and Thank you for your help
Ben

**Comment by jomjol** [comment]
That all looks good. Can you post a serial log during the booting. You can do this with "LOGS & CONSOLE" of the webinstaller

**Comment by HollaDieWaldfee2022** [comment]
Thanks for your reply.
Log without sdcard:

[esp-web-tools-logs (2).txt](https://github.com/jomjol/AI-on-the-edge-device/files/10039310/esp-web-tools-logs.2.txt)

log with sdcard
[esp-web-tools-logs (4).txt](https://github.com/jomjol/AI-on-the-edge-device/files/10039354/esp-web-tools-logs.4.txt)

[esp-web-tools-logs (3).txt](https://github.com/jomjol/AI-on-the-edge-device/files/10039331/esp-web-tools-logs.3.txt)


logs from a second device with the sam sdcard:

**Comment by Aussie-1976** [comment]
You have twice chosen "firmware.bin". In the first row, you need to use "bootloader.bin".

**Reply by HollaDieWaldfee2022** [reply]
Thanks. I corrected that. I did that by accident while preparing the screenshots for the Forum.

![grafik](https://user-images.githubusercontent.com/109627421/202670083-677f9321-3163-48e0-8839-a2ffe962c6c3.png)

But I still get the RAM-Error in the web-logging while starting the device:
E (883) psram: PSRAM ID read error: 0xffffffff
E (883) cpu_start: Failed to init external RAM!


Any other thoughts?

**Reply by jomjol** [reply]
This is not an internal firmware problem. It even does not start the firmware, but stops at initiating the RAM.
Two options:
1) broken bootloader / firmware
2) hardware failure

**Reply by habo20** [reply]
@HollaDieWaldfee2022 , @jomjol 
what was the problem with the error above, I have the same with 2 new ESPCam modules, both modules do not flash the wifi LED after flashing and restarting. Apparently the modules do not boot. Are they then defective?

**Comment by jomjol** [comment]
I don' know, what the problem was / is. As stated above - the very first part of the bootup already fails:

![grafik](https://user-images.githubusercontent.com/30766535/202914106-d46b2f5c-78cf-4c99-80a9-6eae936cb7cd.png)

**Reply by Grish-Lon** [reply]
Same error here, but wrote this off to the PSRAM being "NOT OK" as per  [](https://github.com/jomjol/AI-on-the-edge-device/wiki/Hardware-Compatibility)
bought two different boards - both had NOT OK versions of the PSRAM.  I even zoomed into product photos to see if the PRAM was correct before buying, but supplied boards had a PSRAM change.

Anyone know how to get the right PSRAM chip on this board (I'm in the UK)...

**Reply by Grish-Lon** [reply]
as per Hardware List - see https://github.com/jomjol/AI-on-the-edge-device/wiki/Hardware-Compatibility

**Reply by Grish-Lon** [reply]
I used web-flash (which completed OK, but board would not boot - same error) as well as flash download v3.9.3 with the same results.

**Reply by caco3** [reply]
only the running firmware can check if PSRAM is ok. So the installer will success also on a bad/missing PSRAM!

**Reply by Grish-Lon** [reply]
Reboot-cycle output -

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:7664
load:0x40078000,len:15360
load:0x40080400,len:3744
entry 0x4008067c
I (27) boot: ESP-IDF 4.3.2 2nd stage bootloader
I (27) boot: compile time 20:02:47
I (27) boot: chip revision: 3
I (30) boot_comm: chip revision: 3, min. bootloader chip revision: 0
I (37) boot.esp32: SPI Speed      : 40MHz
I (42) boot.esp32: SPI Mode       : DIO
I (46) boot.esp32: SPI Flash Size : 4MB
I (51) boot: Enabling RNG early entropy source...
I (56) boot: Partition Table:
I (60) boot: ## Label            Usage          Type ST Offset   Length
I (67) boot:  0 nvs              WiFi data        01 02 00009000 00004000
I (74) boot:  1 otadata          OTA data         01 00 0000d000 00002000
I (82) boot:  2 phy_init         RF data          01 01 0000f000 00001000
I (89) boot:  3 ota_0            OTA app          00 10 00010000 001db000
I (97) boot:  4 ota_1            OTA app          00 11 001f0000 001db000
I (104) boot: End of partition table
I (109) boot_comm: chip revision: 3, min. application chip revision: 0
I (116) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=4df30h (319280) map
I (247) esp_image: segment 1: paddr=0005df58 vaddr=3ffb0000 size=020c0h (  8384) load
I (250) esp_image: segment 2: paddr=00060020 vaddr=400d0020 size=15e00ch (1433612) map
I (801) esp_image: segment 3: paddr=001be034 vaddr=3ffb20c0 size=029f8h ( 10744) load
I (806) esp_image: segment 4: paddr=001c0a34 vaddr=40080000 size=1c1f4h (115188) load
I (856) esp_image: segment 5: paddr=001dcc30 vaddr=50000000 size=00010h (    16) load
I (871) boot: Loaded app from partition at offset 0x10000
I (871) boot: Disabling RNG early entropy source...
I (883) psram: This chip is ESP32-D0WD
E (883) psram: PSRAM ID read error: 0xffffffff
E (883) cpu_start: Failed to init external RAM!
Re-enable cpu cache.

abort() was called at PC 0x40082dcf on core 0

Backtrace:0x40082c65:0x3ffe3ba0 0x4008f70d:0x3ffe3bc0 0x40096991:0x3ffe3be0 0x40082dcf:0x3ffe3c50 0x4007956d:0x3ffe3c80 |<-CORRUPTED


ELF file SHA256: 28356ed42df4c694

Rebooting...

**Reply by Grish-Lon** [reply]
Thanks caco3 - that explains when was happening!

Seems difficult to get a board that has the right PSRAM.  Many of the ESP-32-CAM listings do not include the PSRAM information and the photos can be of older product with changes made for current shipping supplies.  I've tried 5 boards so far!

**Reply by caco3** [reply]
That is a pitty!

I ordered 2 from Aliexpress and they have the correct PSRAM. See https://github.com/jomjol/AI-on-the-edge-device/discussions/1289#discussioncomment-4074809

**Reply by Grish-Lon** [reply]
Thanks caco3 - Ordered three of them today.  I'll let you all know how it goes (in a month :P)
